### PR TITLE
[BACKLOG-41091] Decode PVFS File Paths: Scheduler Perspective + Browse Perspective missed case

### PR DESCRIPTION
**Pull request by singletonc**
- Remove erroneous comment from previous PR
- decode pvfs file paths for UI/visually while keeping encoded for backend functionality in New Schedule Dialog and in Schedules Perspective

PR Set:
1. https://github.com/pentaho/pentaho-platform-plugin-common-ui/pull/1777
2. https://github.com/pentaho/pentaho-commons-gwt-modules/pull/1041
3. https://github.com/pentaho/pentaho-platform/pull/5647
4. https://github.com/pentaho/pentaho-scheduler-plugin/pull/183

### CODE REVIEW

**Comment by buildguy** [comment]
### :x: Build failed in 1m 24s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl api,ui
```
---
:ok_hand: All tests passed!
 
Tests run: **98**, Failures: **0**, Skipped: **0** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/pentaho-scheduler-plugin/job/PR-183/1/testReport)

 ---
### Errors:
<details><summary><b>Filtered log</b> (click to expand)</summary>
<p>

```
script returned exit code 1
```
 
</p>
</details>
 
 

:information_source:  _This is an automatic message_

**Review by rmansoor — APPROVED** [review]

File: ui/src/main/java/org/pentaho/mantle/client/dialogs/scheduling/NewScheduleDialog.java [hunk 1]
```diff
@@ -169,7 +172,13 @@ public void setScheduleName( String scheduleName ) {
...
 
   public String getScheduleLocation() {
-    return scheduleLocationTextBox.getText();
+    return partiallyEncodedScheduleLocationPath;
+  }
+  
+  public static void setScheduleLocation( String path ){
+    partiallyEncodedScheduleLocationPath = path;
+    // TODO BACKLOG-41091: Decoding would ideally be coming from the back-end (see also, BACKLOG-41089, BACKLOG-41229)
+    scheduleLocationTextBox.setText( GenericFileNameUtils.urlDecodePvfsFilePath( path ) );
```
Comment (dcleao) [inline ui/src/main/java/org/pentaho/mantle/client/dialogs/scheduling/NewScheduleDialog.java:181]: The output location text box shows a generic file **path**. Paths of VFS files have the `%` character encoded, and the user should see the path unaltered. Not only that, but if, later, this path is further URL-encoded, to be integrated in the path of a URL, or the query string of a URL, or a URL-encoded HTTP body, the `%` character will be encoded twice. Reaching the server, the double encoding gets undone and the original, desired encoding of just the `%` character is restored to the path string. If this does not happen, the path fails to be parsed server-side by the Apache VFS system.
See related story and PR for PDI File Open/Save dialog: https://github.com/pentaho/pentaho-kettle/pull/9450.

The above use case is not the same as that of this story: a label in a folder/file tree/list node, where a _single segment_ of the path is displayed, in which case, individual segments are displayed non-encoded.

If I am right, then no change is needed in this class.

File: ui/src/main/java/org/pentaho/mantle/client/workspace/SchedulesPanel.java [hunk 2]
```diff
@@ -428,7 +429,11 @@ public String getStringValue( JsJob job ) {
       @Override
       public String getStringValue( JsJob jsJob ) {
         try {
-          String outputPath = jsJob.getOutputPath();
+          /*
+          TODO BACKLOG-41091: Decoding would ideally be coming from the back-end (see also, BACKLOG-41089, BACKLOG-41229)
+           we are effectively decoding the value here only to encode it again a few lines down (preventing double-encoding)
+           */
+          String outputPath = GenericFileNameUtils.urlDecodePvfsFilePath( jsJob.getOutputPath() );
```
Comment (dcleao) [inline ui/src/main/java/org/pentaho/mantle/client/workspace/SchedulesPanel.java:436]: Same comment as for `NewScheduleDialog`.

**Review by dcleao — CHANGES_REQUESTED** [review]

**Comment by singletonc** [comment]
Closing in favor of new PR Set:
[pentaho/pentaho-commons-gwt-modules#1042](https://github.com/pentaho/pentaho-commons-gwt-modules/pull/1042)
[pentaho/pentaho-platform#5652](https://github.com/pentaho/pentaho-platform/pull/5652)
[pentaho/pentaho-scheduler-plugin#187](https://github.com/pentaho/pentaho-scheduler-plugin/pull/187)
[pentaho/pentaho-scheduler-plugin-ee#256](https://github.com/pentaho/pentaho-scheduler-plugin-ee/pull/256)

**Pull request closed without merging**